packages/sdk/src/resource-loader.ts

```diff
@@ -89,3 +90,56 @@ export const loadResources = async (
     )
   );
 };
+
+/**
+ * cache api supports only get method
+ * put hash of method and body into url
+ * to support for example graphql queries
+ */
+const getCacheKey = async (request: Request) => {
+  const url = new URL(request.url);
+  const method = request.method;
+  const body = await request.clone().text();
+  // invalidate cache when cache-control is changed
+  const cacheControl = request.headers.get("Cache-Control");
+  const resourceHash = hash(`${method}:${body}:${cacheControl}`);
+  url.searchParams.set("ws-resource-hash", resourceHash);
+  return url;
+};
+
+export const cachedFetch = async (
```

Comment:
does this need a few unit tests? because if we break this we won't easily notice